SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -1128,31 +1128,52 @@ Plotter {
 
 + Function {
 	plot { |duration = 0.01, target, bounds, minval, maxval, separately = false|
+		var server, plotter, action;
+		var name = this.asCompileString;
+
+		if(name.size > 50 or: { name.includes(Char.nl) }) { name = "function plot" };
 
-		var name = this.asCompileString, plotter, action;
-		if(name.size > 50 or: { name.includes(Char.nl) }) { name = "Function plot" };
 		plotter = Plotter(name, bounds);
+		// init data in case function data is delayed (e.g. server booting)
 		plotter.value = [0.0];
+
 		target = target.asTarget;
+		server = target.server;
 		action = { |array, buf|
 			var numChan = buf.numChannels;
-			{
+			var numFrames = buf.numFrames;
+			var frameDur;
+
+			defer {
 				plotter.setValue(
-					array.unlace(numChan).collect(_.drop(-1)),
+					array.unlace(numChan),
```

Comment:
Note: previously, the last data frame was dropped from in the plot. It's not apparent to me why, and seems to work fine plotting _all_ data frames.

Reply:
yes, this seems much better.